mobile/src/navigations/root/root.tsx

```diff
@@ -22,41 +22,56 @@ const screenOptions: NativeStackNavigationOptions = {
 
 const Root: React.FC = () => {
   const dispatch = useAppDispatch();
-  const { isSurveyCompleted, authenticatedUser, authenticatedUserDataStatus } =
-    useAppSelector(({ auth }) => {
-      return {
-        isSurveyCompleted: auth.authenticatedUser?.isSurveyCompleted,
-        authenticatedUser: auth.authenticatedUser,
-        authenticatedUserDataStatus: auth.authenticatedUserDataStatus,
-      };
-    });
+  const {
+    isSurveyCompleted,
+    authenticatedUser,
+    authenticatedUserDataStatus,
+    surveyPreferencesDataStatus,
+  } = useAppSelector(({ auth }) => {
+    return {
+      isSurveyCompleted: auth.authenticatedUser?.isSurveyCompleted,
+      authenticatedUser: auth.authenticatedUser,
+      authenticatedUserDataStatus: auth.authenticatedUserDataStatus,
+      surveyPreferencesDataStatus: auth.surveyPreferencesDataStatus,
+    };
+  });
+
+  const isLoaderVisible =
+    authenticatedUserDataStatus === DataStatus.IDLE ||
+    authenticatedUserDataStatus === DataStatus.PENDING ||
+    surveyPreferencesDataStatus === DataStatus.PENDING;
 
   useEffect(() => {
     void dispatch(authActions.getAuthenticatedUser());
   }, [dispatch]);
 
-  if (
-    authenticatedUserDataStatus === DataStatus.IDLE ||
-    authenticatedUserDataStatus === DataStatus.PENDING
-  ) {
-    return <Loader />;
-  }
-
   return (
-    <NativeStack.Navigator screenOptions={screenOptions}>
-      {authenticatedUser ? (
-        isSurveyCompleted ? (
-          <NativeStack.Screen name={RootScreenName.MAIN} component={Main} />
+    <>
+      <NativeStack.Navigator screenOptions={screenOptions}>
+        {authenticatedUser ? (
+          isSurveyCompleted ? (
+            <NativeStack.Screen name={RootScreenName.MAIN} component={Main} />
+          ) : (
+            <NativeStack.Screen
+              name={RootScreenName.SURVEY}
+              component={Survey}
+            />
+          )
         ) : (
-          <NativeStack.Screen name={RootScreenName.SURVEY} component={Survey} />
-        )
-      ) : (
-        <>
-          <NativeStack.Screen name={RootScreenName.SIGN_IN} component={Auth} />
-          <NativeStack.Screen name={RootScreenName.SIGN_UP} component={Auth} />
-        </>
-      )}
-    </NativeStack.Navigator>
+          <>
+            <NativeStack.Screen
+              name={RootScreenName.SIGN_IN}
+              component={Auth}
+            />
+            <NativeStack.Screen
+              name={RootScreenName.SIGN_UP}
+              component={Auth}
+            />
+          </>
+        )}
+      </NativeStack.Navigator>
+      <Loader isVisible={isLoaderVisible} />
+    </>
```

Comment:
it's not a good approach of nesting ternary operator, it increases the time to read and figure out, let's create a helper which will receive conditions and return available screens to render